Source/missiles.cpp

```diff
@@ -200,22 +200,24 @@ bool MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, missile_id t
 	if (monster._mmode == MonsterMode::Charge)
 		return false;
 
-	uint8_t mor = monster.mMagicRes;
+	MonsterResists mor = monster.mResists;
 	missile_resistance mir = MissilesData[t].mResist;
 
-	if (((mor & IMMUNE_MAGIC) != 0 && mir == MISR_MAGIC)
-	    || ((mor & IMMUNE_FIRE) != 0 && mir == MISR_FIRE)
-	    || ((mor & IMMUNE_LIGHTNING) != 0 && mir == MISR_LIGHTNING)
-	    || ((mor & IMMUNE_ACID) != 0 && mir == MISR_ACID))
+	if ((mor.isMagicImmune() && mir == MISR_MAGIC) || (mor.isFireImmune() && mir == MISR_FIRE) || (mor.isLightningImmune() && mir == MISR_LIGHTNING) || (mor.isAcidImmune() && mir == MISR_ACID)) {

```

Comment:
Rather then having several  functions I think it would be a lot cleaner to have an element enum and have `Resist::isImmune()` take it as a parameter. Else anyone who want to expand on the type will have to chance all lines like this down and add to them.